### PR TITLE
Fix External should also be ProviderID

### PR DESCRIPTION
**Pull request by kebe7jun**
External should also be ProviderID type

### CODE REVIEW

File: pilot/pkg/networking/apigen/apigen.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func (g *APIGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w *
 		svcs := push.Services(proxy)
 		for _, s := range svcs {
 			// Ignore services that are result of conversion from ServiceEntry.
-			if s.Attributes.ServiceRegistry == serviceregistry.External {
+			if s.Attributes.ServiceRegistry == string(serviceregistry.External) {
```
Comment (devincd) [inline pilot/pkg/networking/apigen/apigen.go:145]: I think let `s.Attributes.ServiceRegistry` to be `ProviderID` type is more better?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/apigen/apigen.go:145]: good suggestion

Comment (kebe7jun) [reply pilot/pkg/networking/apigen/apigen.go]: thanks, fixed.

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by kebe7jun** [comment]
@shamsher31 can you please take a look here?

**Review by shamsher31 — APPROVED** [review]
Thanks

**Pull request merged**